lib/Model/Listing/Dao/QueryBuilderHelperTrait.php

```diff
@@ -120,45 +112,41 @@ private function applyOrderByToQueryBuilder(QueryBuilder $queryBuilder): void
         }
     }
 
-    /**
-     * @param DoctrineQueryBuilder $queryBuilder
-     */
     private function applyLimitToQueryBuilder(QueryBuilder $queryBuilder): void
     {
         $queryBuilder->setFirstResult($this->model->getOffset());
         $queryBuilder->setMaxResults($this->model->getLimit());
     }
 
-    protected function prepareQueryBuilderForTotalCount(DoctrineQueryBuilder &$queryBuilder): void
+    protected function prepareQueryBuilderForTotalCount(QueryBuilder $queryBuilder, string $identifierColumn): void
```

Comment:
Does this work without passing as reference? as nothing is returned from the method.

Reply:
Yes, the QueryBuilder object itself is updated (instead of updating the argument value to a string which was the previous approach)

```
$queryBuilder
                ->resetQueryParts()
                ->select('COUNT(*)')
                ->from('(' . $innerQuery . ')', 'XYZ')
            ;
```